skill_tagging/utils.py

```diff
@@ -0,0 +1,18 @@
+"""
+Utility functions for SkillTaggingMixin
+"""
+import requests
+
+
+def get_api_client(user):
+    """
+    Get the authenticated api client for user
+    """
+    # pylint: disable=import-error,import-outside-toplevel
+    from edx_rest_api_client.auth import SuppliedJwtAuth
+    from openedx.core.djangoapps.oauth_dispatch.jwt import create_jwt_for_user
+
+    client = requests.Session()
+    jwt = create_jwt_for_user(user)
```

Comment:
@pkulkark I am facing an error while trying to call `self._fetch_skill_tags()` inside vertical_block.py

`'XBlockUser' object has no attribute 'username'`

```
edx.devstack.lms  |   File "/edx/app/edxapp/edx-platform/xmodule/vertical_block.py", line 189, in student_view
edx.devstack.lms  |     return self._student_or_public_view(context, STUDENT_VIEW)
edx.devstack.lms  |   File "/edx/app/edxapp/edx-platform/xmodule/vertical_block.py", line 99, in _student_or_public_view
edx.devstack.lms  |     print(self._fetch_skill_tags())
edx.devstack.lms  |   File "/edx/src/xblock-skill-tagging/skill_tagging/skill_tagging_mixin.py", line 55, in _fetch_skill_tags
edx.devstack.lms  |     api_client = get_api_client(user=user)
edx.devstack.lms  |   File "/edx/src/xblock-skill-tagging/skill_tagging/utils.py", line 17, in get_api_client
edx.devstack.lms  |     jwt = create_jwt_for_user(user)
edx.devstack.lms  |   File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/oauth_dispatch/jwt.py", line 39, in create_jwt_for_user
edx.devstack.lms  |     return _create_jwt(
edx.devstack.lms  |   File "/edx/app/edxapp/edx-platform/openedx/core/djangoapps/oauth_dispatch/jwt.py", line 175, in _create_jwt
edx.devstack.lms  |     'preferred_username': user.username,
edx.devstack.lms  | AttributeError: 'XBlockUser' object has no attribute 'username'
```

So I checked contents of `user` variable:
```
edx.devstack.lms  | ======================================= get_api_client user: {'is_current_user': True, 'emails': ['edx@example.com'], 'full_name': '', 'opt_attrs': {'edx-platform.anonymous_user_id': '3bbcd288d784095b82f549903a9fe00e', 'edx-platform.is_authenticated': True, 'edx-platform.request_country_code': None, 'edx-platform.user_id': 3, 'edx-platform.username': 'edx', 'edx-platform.user_is_staff': True, 'edx-platform.user_role': 'instructor', 'edx-platform.user_preferences': {}}}
```

My guess is that we need to fetch user from database as `create_jwt_for_user` expects an instance of user model.
Some useful links:
- https://github.com/openedx/XBlock/blob/c3e002933c8380e38599e000effa0121d600b039/xblock/reference/user_service.py#L23
- https://github.com/openedx/edx-platform/blob/master/common/djangoapps/xblock_django/user_service.py#L89

Reply:
@navinkarkera Oh I see. I've updated it to use the `get_user_by_anonymous_id` method. Can you check if that works? My devstack is borked again!

Reply:
@pkulkark Nice! it works.